File: Parsimmon/Example/ClassifierViewController.m
//
//  ClassifierViewController.m
//  Parsimmon
//
//  Created by Ayaka Nonaka on 10/13/13.
//
//

#import "ClassifierViewController.h"
#import "Parsimmon.h"
#import "Parsimmon-Swift.h"

@interface ClassifierViewController ()
@property (weak, nonatomic) IBOutlet UITextField *messageTextField;
@property (weak, nonatomic) IBOutlet UILabel *resultLabel;
@property (strong, nonatomic) NaiveBayesClassifier *classifier;
@end

@implementation ClassifierViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	[self train];
}


#pragma mark - Training

- (void)train
{
    NSArray *hams = @[@"Hi, how was work?",
                      @"What's the weather like tomorrow?",
                      @"Please take the dogs out for a walk."];
    NSArray *spams = @[@"Get your free copy of AOL today!",
                       @"Get a free pink iPhone by clicking HERE",
                       @"Spam spam spam spam."];
    [self feedHams:hams];
    [self feedSpams:spams];
}

- (void)feedHams:(NSArray *)hams
{
    for (NSString *ham in hams) {
        [self.classifier trainWithText:ham category:@"ham"];
    }
}

- (void)feedSpams:(NSArray *)spams
{
    for (NSString *spam in spams) {
        [self.classifier trainWithText:spam category:@"spam"];
    }
}


#pragma mark - Actions

- (IBAction)spamOrHamAction:(id)sender
{
    NSString *category = [self.classifier classify:self.messageTextField.text];
    [self.resultLabel setText:category];
}


#pragma mark - Properties

- (NaiveBayesClassifier *)classifier
{
    if (!_classifier) {
        _classifier = [[NaiveBayesClassifier alloc] init];
    }
    return _classifier;
}


@end

File: Parsimmon/Example/ClassifierViewController.swift
//
//  ClassifierViewController.swift
//  Parsimmon
//
//  Created by Ayaka Nonaka on 10/13/13.
//
//

import Foundation
import UIKit

class ClassifierViewController: UIViewController {
    @IBOutlet private weak var messageTextField: UITextField!
    @IBOutlet private weak var resultLabel: UILabel!
    
    private lazy var classifier = NaiveBayesClassifier()

    @IBAction private func spamOrHamAction(sender: UIButton) {
        let category = classifier.classify(messageTextField.text)
        resultLabel.text = category
    }

    private func train() {
        let hams = [
            "Hi, how was work?",
            "What's the weather like tomorrow?",
            "Please take the dogs out for a walk."
        ]
        let spams = [
            "Get your free copy of AOL today!",
            "Get a free pink iPhone by clicking HERE",
            "Spam spam spam spam."
        ]
        feedHams(hams)
        feedSpams(spams)
    }

    private func feedHams(hams: [String]) {
        for ham in hams {
            classifier.trainWithText(ham, category: "ham")
        }
    }

    private func feedSpams(spams: [String]) {
        for spam in spams {
            classifier.trainWithText(spam, category: "spam")
        }
    }
}

extension ClassifierViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        train()
    }
}
